pkg/gofr/cmd/terminal/progress_test.go

```diff
@@ -0,0 +1,131 @@
+package terminal
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gofr.dev/pkg/gofr/testutil"
+)
+
+func TestProgressBar_SuccessCases(t *testing.T) {
+	total := int64(100)
+
+	var out bytes.Buffer
+	stream := &Output{terminal{isTerminal: true, fd: 1}, &out}
+	bar := NewProgressBar(stream, total)
+
+	// Mock terminal size
+	bar.tWidth = 120
+
+	// Increment the progress bar
+	bar.Incr(10)
+
+	// Verify the output
+	expectedOutput := "\r[████░                                             ] 10.000%"
+	if out.String() != expectedOutput {
+		t.Errorf("Unexpected progress bar output: got %q, want %q", out.String(), expectedOutput)
+	}
+
+	// Increment the progress bar to completion
+	bar.Incr(total - 10)
+
+	// Verify the completion output
+	expectedCompletion := "\r[████░                                             ] 10.000%\r" +
+		"[██████████████████████████████████████████████████] 100.000%\n"
+	if out.String() != expectedCompletion {
+		t.Errorf("Unexpected completion output: got %q, want %q", out.String(), expectedCompletion)
+	}
+}
+
+func TestProgressBar_Fail(t *testing.T) {
+	out := testutil.StdoutOutputForFunc(func() {
+		var out bytes.Buffer
+		stream := &Output{terminal{isTerminal: true, fd: 1}, &out}
+		bar := NewProgressBar(stream, int64(-1))
+
+		assert.Equal(t, bar.total, int64(0))
```

Comment:
Expected and actual are inverted no?


```suggestion
		assert.Equal(t, int64(0), bar.total)
```

But I would use Zero anyway


```suggestion
		assert.Zero(t, bar.total)
```